gpiozero/boards.py

```diff
@@ -1156,6 +1156,8 @@ def close(self):
         if self._master:
             m = self._master
             self._master = None
+            self._socket = None
+            self._value = None
```

Comment:
... so looking at https://github.com/RPi-Distro/python-gpiozero/blob/master/gpiozero/input_devices.py#L200 perhaps `self._socket` *shouldn't* be set to `None`, since I guess there's no harm in knowing what socket it *used* to be connected to? Whereas trying to query or set the value should definitely raise `DeviceClosed`.